src/Uno.UI.RuntimeTests/UITests/_Engine/QueryExtensions.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Windows.Foundation;
+using Windows.UI.Xaml;
+using Uno.UITest;
+
+namespace Uno.UITest.Helpers.Queries;
+
+internal static class QueryExtensions
+{
+	public static IEnumerable<QueryResult> WaitForElement(this SkiaApp app, string elementName)
+		=> new QueryEx(app, elementName).Execute();
+
+	public static QueryEx Marked(this SkiaApp app, string elementName)
+		=> new(app, elementName);
+
+	public static void FastTap(this QueryEx query)
+	{
+		var bounds = query.Execute().Single().Rect;
+
+		query.App.TapCoordinates((float)(bounds.X + bounds.Width / 2), (float)(bounds.Y + bounds.Height / 2));
```

Comment:
Are you sure it will work all the time, considering some screen pixel density...

Reply:
It should since we are dealing only with virtual pixels here, Anyway it's only a preliminary work, we will fix later if needed.